crates/forge/src/scarb/config.rs

```diff
@@ -0,0 +1,127 @@
+use anyhow::{anyhow, Result};
+use conversions::StarknetConversions;
+use itertools::Itertools;
+use serde::Deserialize;
+use starknet::core::types::{BlockId, BlockTag};
+use std::collections::HashMap;
+use url::Url;
+
+#[derive(Debug, PartialEq, Default)]
+#[allow(clippy::module_name_repetitions)]
+pub struct ForgeConfig {
+    /// Should runner exit after first failed test
+    pub exit_first: bool,
+    /// How many runs should fuzzer execute
+    pub fuzzer_runs: Option<u32>,
+    /// Seed to be used by fuzzer
+    pub fuzzer_seed: Option<u64>,
+
+    pub fork: Vec<ForkTarget>,
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct ForkTarget {
+    pub name: String,
+    pub url: Url,
+    pub block_id: BlockId,
+}
+
+/// Represents forge config deserialized from Scarb.toml using basic types like String etc.
+#[derive(Deserialize, Debug, PartialEq, Default)]
+pub(crate) struct RawForgeConfig {
+    #[serde(default)]
+    /// Should runner exit after first failed test
+    pub exit_first: bool,
+    /// How many runs should fuzzer execute
+    pub fuzzer_runs: Option<u32>,
+    /// Seed to be used by fuzzer
+    pub fuzzer_seed: Option<u64>,
+
+    #[serde(default)]
+    pub fork: Vec<RawForkTarget>,
+}
+
+#[derive(Deserialize, Debug, PartialEq, Default, Clone)]
+pub(crate) struct RawForkTarget {
+    pub name: String,
+    pub url: String,
+    pub block_id: HashMap<String, String>,
+}
+
+pub(super) fn validate_raw_fork_config(raw_config: &RawForgeConfig) -> Result<()> {
+    let forks = &raw_config.fork;
+    let names: Vec<String> = forks.iter().map(|fork| fork.name.clone()).collect();
+    let removed_duplicated_names: Vec<String> = names.clone().into_iter().unique().collect();
+
+    if names.len() != removed_duplicated_names.len() {
+        return Err(anyhow!("Some fork names are duplicated"));
+    }
+
+    for fork in forks {
+        let block_id_items: Vec<(&String, &String)> = fork.block_id.iter().collect();
+        let [(block_id_key, block_id_value)] = block_id_items[..] else {
+            return Err(anyhow!("block_id should be set once per fork"));
+        };
+
+        if !["number", "hash", "tag"].contains(&&**block_id_key) {
+            return Err(anyhow!(
+                "block_id has only three variants: number, hash and tag"
+            ));
```

Comment:
This errors should be structured more like `"block_id = {.. user input ..} is not valid. Possible values = are "number", "hash" and "tag"`. 

With errors like this, it's hard for the users to find where exactly in the fork configuration the error was made.